lzss/compress.go

```diff
@@ -246,7 +202,7 @@ func (compressor *Compressor) write(w writer, d []byte, startIndex int, inputInd
 	for i := startIndex; i < len(d); {
 		// if we have a series of repeating bytes, we can do "RLE" using a short backref
 		count := 0
-		for i+count < len(d) && count < (1<<maxBackrefLen) && d[i] == d[i+count] {
+		for i+count < len(d) && count < (1<<maxBackrefLenLog2) && d[i] == d[i+count] {
```

Comment:
I would add a comment here to explain why it's okay to have a constant loop bound rather than a dynamic one.